### ISSUE TITLE
New Feature: "Projected Collections"

### ISSUE THREAD

**Issue by SvenEwald** [SvenEwald/xmlbeam#33]
Projection methods could return a new type "ProjectedList<T> extends List<T>".
Changes to the this list could directly change the underlying document.
This could reduce some code like

```
List<> list = projection.getList();
list.add(xx)
projection.setList(list)
```

to
`projection.getList().add(xx)`
Maybe there are more collection types that could benefit from this idea.

**Comment by SvenEwald** [comment]
Feature works pretty well with lists. Digging into a Map implementation now...

**Comment by SvenEwald** [comment]
Using Automaps with empty documents does not work, yet. Need to wait until 1.4.12.

**Comment by SvenEwald** [comment]
included in 1.4.12
